test/test_torch.py

```diff
@@ -4849,6 +4849,54 @@ def test_diff(self, device, dtype):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
 
+    def _test_histogram_numpy(self, t, bins, weights, density):
+        # Helper for test_histogram to compare with NumPy reference implementation
+        def to_np(t):
+            if t is None:
+                return None
+            if t.dtype == torch.bfloat16:
+                return t.to(dtype=torch.float, device="cpu").numpy()
+            else:
+                return t.cpu().numpy()
+
+        np_t = to_np(t)
+        np_bins = bins if isinstance(bins, int) else to_np(bins)
```

Comment:
If the above change to `to_np` is accepted then I think this can unconditionally `to_np` bins